### PR TITLE
fix issue #344, but can not pass test

### PR DESCRIPTION
**Pull request by lornally**
maybe deep, static, property and method clone need fix.

### CODE REVIEW

File: src/stampit.js [hunk 1]
```diff
@@ -46,7 +46,16 @@
     return dst;
   }
 
-  var assign = _Object.assign || _mergeOrAssign.bind(0, assignOne);
+ // var assign = _Object.assign || _mergeOrAssign.bind(0, assignOne);
+ var assign = (ob, ...o) => {
+  o.forEach((obj) => {
+    if (typeof obj !== 'undefined') {
+      Object.defineProperties(ob, Object.getOwnPropertyDescriptors(obj));
```
Comment (koresar) [inline src/stampit.js]: I believe all we need to do is to insert this line after  the line 44. And it will just work I think. Try that.

**Comment by lornally** [comment]
ok, i will try it.

File: src/stampit.js (hunk 1 above)
Comment (lornally) [reply src/stampit.js]: i try this, but stil can not pass test:
```js
function _mergeOrAssign(action, dst) {
    return slice.call(arguments, 2).reduce(action, dst);
  }

  function assignOne(dst, src) {
    if (src) {
      // var keys = objectKeys(src);
      // for (var i = 0; i < keys[_length]; i++) {
      //   dst[keys[i]] = src[keys[i]];
			// }
			Object.defineProperties(dst, Object.getOwnPropertyDescriptors(src));

    }
    return dst;
  }

 // var assign = _Object.assign || _mergeOrAssign.bind(0, assignOne);
 var assign = (ob, ...o) => {
  o.forEach((obj) => {
    if (typeof obj !== 'undefined') {
      Object.defineProperties(ob, Object.getOwnPropertyDescriptors(obj));
    }
  });

  return ob;
};

  function isFunction(obj) {
    return typeof obj == 'function';
  }
```

Comment (lornally) [reply src/stampit.js]: i got it, thank you very much:)

**Comment by lornally** [comment]
i have push the pr, and,  because the eslint error, so i do a prettier format, sorry.

**Comment by koresar** [comment]
No worries mate. All good. I'll take a look later.

**Comment by lornally** [comment]
sorry there is a format error, too.

**Comment by lornally** [comment]
sorry, mate, there is error still,  and i can not deal it. help :)

**Comment by koresar** [comment]
I will take a look today or tomorrow. 
Very busy at work currently.
Sorry about that

**Comment by lornally** [comment]
you are welcome.  and ask a digression from this issue, how can you react so in time? you can receive some github message? or maybe we can contact each other with some IM tools.

**Comment by koresar** [comment]
@lornally I do receive emails for each of the issue, PR or comment. :)
You can reach me on twitter https://twitter.com/kore_sar

**Comment by lornally** [comment]
thank you, i will set github so i can receive the mail too.  btw, the pr can pass npm test, but can not pass CI.

**Comment by lornally** [comment]
hi, mate, can we continue?

**Comment by koresar** [comment]
Yes we can. But after this one merged.
https://github.com/stampit-org/stamp-specification/pull/128

The specification is a crucial part of the project. Every module which
implements the stamp specification must fully comply.

Thus I believe it would be a good idea to wait for the next version of the
specification.

Sorry I didn't mention it earlier

**Comment by lornally** [comment]
ok, no problem, you are welcome, let us do it, we can build the specification now:)

**Comment by koresar** [comment]
Implemented with #346

**Pull request closed without merging**